types/ember-data/index.d.ts

```diff
@@ -23,7 +23,30 @@ import AdapterRegistry from 'ember-data/types/registries/adapter';
 type ModelKeys<Model extends DS.Model> = Exclude<keyof Model, keyof DS.Model>;
 
 type AttributesFor<Model extends DS.Model> = ModelKeys<Model>; // TODO: filter to attr properties only (TS 2.8)
-type RelationshipsFor<Model extends DS.Model> = ModelKeys<Model>; // TODO: filter to hasMany/belongsTo properties only (TS 2.8)
+type RelationshipKeysFor<Model extends DS.Model> = keyof RelationshipsFor<Model>;
+
+type RelationshipsFor<Model extends DS.Model> = HasManyRelationshipsFor<Model> & BelongsToRelatipsFor<Model>;
+
+type BelongsToRelatipsFor<Model extends DS.Model> = {
+    [k in keyof Model]:
+      Model[k] extends Ember.ComputedProperty<infer C & DS.PromiseObject<infer C>, infer C>
+        ? C extends DS.Model ? C : never
+      : Model[k] extends DS.Model ? Model[k]
```

Comment:
this line stands to handle decorators version. Should probably add 
```ts
: Model[k] extends DS.PromiseObject<infer C> ? C
```
to support 
```ts
export default class Post extends Model {
  @belongsTo('user')
  user!: DS.PromiseObject<User>;
}
```